internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -566,6 +566,13 @@ func isUserManagedNetwork(clusterInstall *hiveext.AgentClusterInstall) bool {
 	return false
 }
 
+func isDiskEncryptionEnabled(clusterInstall *hiveext.AgentClusterInstall) bool {
+	return clusterInstall.Spec.DiskEncryption != nil &&
+		(swag.StringValue(clusterInstall.Spec.DiskEncryption.EnableOn) == models.DiskEncryptionEnableOnAll ||
+			swag.StringValue(clusterInstall.Spec.DiskEncryption.EnableOn) == models.DiskEncryptionEnableOnMasters ||
+			swag.StringValue(clusterInstall.Spec.DiskEncryption.EnableOn) == models.DiskEncryptionEnableOnWorkers)
+}
+
```

Comment:
```suggestion
return clusterInstall.Spec.DiskEncryption != nil &&
 		swag.StringValue(clusterInstall.Spec.DiskEncryption.EnableOn) != models.DiskEncryptionEnableOnNone 
```

Reply:
I'd rather test the specific positive cases, in case we may have some bad / empty values in here